### DISCUSSION TITLE
Input Variable Bounds

### DISCUSSION THREAD

**Discussion by egary1** [bitcraze/discussions#1410] [Q&A]
I am currently trying to take data on frequency sweeps at hover for the compatible flapper drone, and I am running into issues with some of the command (input) variables. The bounds of these 16 bit variables were being hit during the sweeps, so I lowered the attitude gains for the onboard PID controller. This allowed most of the inputs to decrease by a tolerable margin, letting me take data without hitting the bounds of the 16-bit format.

However, for the yaw command variable I am still hitting the bounds even with a miniscule proportional gain with the controller (I have lowered it all the way to .10 with no Integral or derivative control. 

I'm wondering if this 16-bit bound is an arbitrary limit of the hardware itself (meaning the controller just sends the computer information as 16-bit numbers and could be providing inputs beyond what the computer can comprehend), or if the bound represents actuator saturation (max input) that the controller can possibly provide, meaning that the bounds are max possible inputs in each respective direction. 

Any help would be greatly appreciated.

**Comment by knmcguire** [comment]
@matejkarasek could you perhaps answer this?

**Reply by matejkarasek** [reply]
Sorry, only see this now.

I already replied directly to @egary1, the question was about the cmd_pitch, cmd_roll and cmd_yaw variables. 
These variable are indeed 16bit and on the flapper drone do occasionally saturate (high frequency vibrations, amplified by the PID gains).
These same, saturated values are used in the power distribution, where the motor/servoactuator commands are computed. So if you log cmd_pitch, etc., these are the signals actually used. The actuator/servo will only be able to follow the low frequency content of these signals, ignoring the occasional, saturated, spikes.
